dashboard/src/components/Navbar.tsx

```diff
@@ -76,14 +79,16 @@ export default function Navbar({ userPic }: { userPic: string }) {
                   id='darkmodelightmodetoggle'
                   className='h-8 w-8'
                   onClick={() => {
-                    const newIsDarkMode = !isDarkMode;
-                    localStorage.setItem('theme', newIsDarkMode ? 'dark' : 'light');
-                    if (newIsDarkMode) {
-                      document.documentElement.classList.add('dark');
-                    } else {
-                      document.documentElement.classList.remove('dark');
-                    }
-                    setDarkMode(newIsDarkMode);
+                    (() => {
```

Comment:
nit - why do we need this function nesting?

Reply:
I do not fully understand why, but making it an IIFE seems to resolve the "first click glitch" on the darkmode/lightmode toggle.

Reply:
oh I never moved the `isDarkMode` to app state... So probably the `setDarkMode` was the one causing the rendering... Some tech debt to be cleaned up later

Reply:
Hmm. I don't think so. Or at least, it's not directly related to this bug here, as this one only triggers when you click on the icon to toggle dark mode and light mode.